sim/src/pandemic.rs

```diff
@@ -135,36 +182,155 @@ impl PandemicModel {
         // person has spent some duration in the same space as other people. Does transmission
         // occur?
         for (other, overlap) in other_occupants {
-            if self.infected.contains(&person) != self.infected.contains(&other) {
-                if overlap > Duration::hours(1) && self.rng.gen_bool(0.1) {
-                    if self.infected.contains(&person) {
-                        self.become_infected(now, other, scheduler);
-                    } else {
-                        self.become_infected(now, person, scheduler);
-                    }
+            if let Some(pid) = self.might_become_exposed(person, other) {
+                if self.exposition_occurs(overlap) {
+                    self.become_exposed(now, pid, scheduler);
                 }
             }
         }
     }
 
-    fn become_infected(&mut self, now: Time, person: PersonID, scheduler: &mut Scheduler) {
-        self.infected.insert(person);
+    // transition from a state to another without interaction with others
+    fn transition(&mut self, now: Time, person: PersonID, scheduler: &mut Scheduler) {
+        // person has spent some duration in the same space as other people. Does transmission
+        // occur?
+        let inf_pers = self.infected.get(&person).map(|pers| *pers);
+        if let Some((t0, last_check)) = inf_pers {
```

Comment:
```suggestion
        if let Some((t0, last_check)) = self.infected.get(&person).cloned() {
```
Slightly more readable? Same below